kernel/kernel-api/src/main/java/io/delta/kernel/internal/TableImpl.java

```diff
@@ -90,19 +93,39 @@ public String getPath(Engine engine) {
 
   @Override
   public Snapshot getLatestSnapshot(Engine engine) throws TableNotFoundException {
-    return snapshotManager.buildLatestSnapshot(engine);
+    SnapshotQueryContext snapshotContext = SnapshotQueryContext.forLatestSnapshot(tablePath);
```

Comment:
Just curious (sorry if I asked this before the winter holiday's and then just forgot the answer):

Any reason why we do this try and catch inside `TableImpl::getLatestSnapshot` instead of `SnapshotManager::buildLatestSnapshot` ?

Reply:
I think it's probably just a matter of preference. If we do it in SnapshotManager we still have to do it for all 3 access patterns.

I think I put it here since it makes sense to do it at the outermost layer and catch any exceptions thrown by the nested functions.